### ISSUE TITLE
空の DataLoader でのゼロ除算回避

### ISSUE THREAD

**Issue by kurorosu** [kurorosu/pochitrain#136]
## 概要

- `pochitrain/pochi_trainer.py` と `pochitrain/training/evaluator.py` で `len(loader)` と `total` に依存した平均計算
- `len(loader) == 0` または `total == 0` で例外発生

### 1 ガード追加

- `len(loader)` と `total` が 0 の場合の分岐
- サンプルなし時の戻り値定義

## コード変更例

```python
# pochitrain/pochi_trainer.py
avg_loss = total_loss / total if total > 0 else 0.0
```

## 対象ファイル

- `pochitrain/pochi_trainer.py` — 学習ループの平均計算とガード追加
- `pochitrain/training/evaluator.py` — 評価の平均計算とガード追加
- `tests/unit/test_core/test_pochi_trainer.py` — 空 DataLoader のケース追加
- `tests/unit/test_core/test_evaluator.py` — 空 DataLoader のケース追加

## Labels

`bug`

## Branch

`fix/empty-dataloader-guards`
